src/Export/ExportService.php

```diff
@@ -206,20 +207,26 @@ public function allRegistrationDataToCSV(Event $event, User $adminUser): array
         ];
 
         foreach ($participants as $participant) {
-            if ($participant instanceof PatrolLeader) {
-                $pPart = [
+            if ($participant instanceof PatrolLeader || $participant instanceof TroopLeader) {
+                $ptPart = [
                     (string)$participant->id,
                     $participant->patrolName ?? '',
                     (string)$participant->getPatrolParticipantsCount(),
                 ];
             } elseif ($participant instanceof PatrolParticipant) {
-                $pPart = [
+                $ptPart = [
                     (string)$participant->patrolLeader->id,
                     $participant->patrolLeader->patrolName ?? '',
                     '',
                 ];
+            } elseif ($participant instanceof TroopParticipant) {
+                $ptPart = [
+                    (string)$participant->troopLeader->id,
```

Comment:
tady to může zhasnout, protože atribut troopLeader může být null - máme i volné troop participanty, kteří ještě nemají troop -> tedy nemají ani troop leadera